src/mpi/comm/comm_impl.c

```diff
@@ -678,6 +678,72 @@ int MPIR_Comm_dup_with_info_impl(MPIR_Comm * comm_ptr, MPIR_Info * info, MPIR_Co
     goto fn_exit;
 }
 
+int MPIR_Comm_create_from_group_impl(MPIR_Group * group_ptr, const char *stringtag,
+                                     MPIR_Info * info_ptr, MPIR_Errhandler * errhan_ptr,
+                                     MPIR_Comm ** p_newcom_ptr)
+{
+    int mpi_errno = MPI_SUCCESS;
+
+    /* This implementation assumes an internal MPI_COMM_WORLD already exist.
+     *
+     * Refer to TODOs below inside the branches:
+     * The next implementation will relax but assume the first call to this function will
+     * either from a "world" pset or "self" pset, during former a comm world will be created.
+     *
+     * The final implementation will further relax and allow partial initializations.
+     */
+
+    /* Need convert stringtag to tag, use macro from uthash.h to calculate hash value */
+    unsigned hash;
+    int n = strlen(stringtag);
+    HASH_VALUE(stringtag, n, hash);
+
+    /* NOTE: tag will be used with MPIR_TAG_COLL_BIT on, ref. MPIR_Get_contextid_sparse_group */
+    int tag = hash % (MPIR_Process.attrs.tag_ub);
+
```

Comment:
ref: https://github.com/pmodels/mpich/pull/4949/commits/accb10b2756f1a55619eb2fd7b4fb6fdbf9df5fe

Reply:
Respond to
> It is only two lines and it is only used here, making it a separate routine seems unnecessary. Do I need improve the comments?

The logic here is clear but just I feel that the logic is standalone, so might be more readable if we define a subroutine for it (not for reuse).